query/src/sessions/context_shared.rs

```diff
@@ -71,8 +72,15 @@ impl QueryContextShared {
         conf: Config,
         session: Arc<Session>,
         cluster_cache: Arc<Cluster>,
-    ) -> Arc<QueryContextShared> {
-        Arc::new(QueryContextShared {
+    ) -> Result<Arc<QueryContextShared>> {
+        let dal_ctx = DalContext::create(
```

Comment:
@zhang2014  need your help :)

Is it possible that there might be more than one instance of `QueryContextShared` co-existing in a single instance of query node?

the concern is :

`DalContext::create` here will also create a new instance of DalCache (if the cache is enabled), which uses the same disk path as cache dir. if multiple instances of `DalCache` write to the same file (and the file is large enough), we might read partially written cache.

Reply:
> Is it possible that there might be more than one instance of QueryContextShared co-existing in a single instance of query node?

No. Maybe we should move to session manager?

Reply:
got it, thanks!

> Maybe we should move to SessionManager(single instance for single query node)?

yeah, I have the same feelings

Reply:
@dantengsky ok, I'll put it in SessionManager.